catalog/app/containers/Admin/AddMember.js

```diff
@@ -0,0 +1,140 @@
+import FlatButton from 'material-ui/FlatButton';
+import TextField from 'material-ui/TextField';
+import { red500 } from 'material-ui/styles/colors';
+import PT from 'prop-types';
+import React, { Fragment } from 'react';
+import { FormattedMessage as FM, injectIntl } from 'react-intl';
+import {
+  compose,
+  setDisplayName,
+  setPropTypes,
+} from 'recompose';
+import { reduxForm, Field, SubmissionError } from 'redux-form/immutable';
+import styled from 'styled-components';
+
+import { push } from 'containers/Notifications/actions';
+import Spinner from 'components/Spinner';
+import * as validators from 'utils/validators';
+
+import msg from './messages';
+
+const FormField = compose(
+  setPropTypes({
+    input: PT.object.isRequired,
+    meta: PT.shape({
+      submitting: PT.bool,
+      submitFailed: PT.bool,
+      error: PT.string,
+    }).isRequired,
+    errors: PT.objectOf(PT.string),
+  }),
+  setDisplayName('Admin.AddMember.FormField'),
+// eslint-disable-next-line object-curly-newline
+)(({ input, meta, errors = {}, ...props }) => (
+  <TextField
+    {...input}
+    {...props}
+    disabled={meta.submitting}
+    errorText={meta.submitFailed ? errors[meta.error] || meta.error : undefined}
+  />
+));
+
+const Form = styled.form`
+  align-items: baseline;
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const FormError = styled.p`
+  color: ${red500};
+  flex-basis: 100%;
+  font-family: Roboto !important;
+  font-size: .8em;
+`;
+
+const FORM_ERRORS = {
+  uniq: msg.addMemberFormErrorUniq,
+  username: msg.addMemberFormErrorUsername,
+};
+
+export default compose(
+  injectIntl,
+  setPropTypes({
+    addMember: PT.func.isRequired,
+    intl: PT.shape({
+      formatMessage: PT.func.isRequired,
+    }).isRequired,
+  }),
+  reduxForm({
+    form: 'Admin.AddMember',
+    onSubmit: (values, _dispatch, { addMember }) =>
+      new Promise((resolve, reject) => addMember(values.toJS(), { resolve, reject }))
+        .catch((err) => {
+          if (err.message === "Bad request. Maybe there's already a user with the username you provided?") {
```

Comment:
minor; why `"` and `'` elsewhere?

Reply:
single quotes are the default ones (even enforced by the linter), but here we have a single quote inside a string, so we use double quotes to avoid escaping